src/bun.js/api/server/SSLConfig.zig

```diff
@@ -227,6 +227,89 @@ pub fn deinit(this: *SSLConfig) void {
         this.ca = null;
     }
 }
+pub fn clone(this: *const SSLConfig) SSLConfig {
+    var cloned: SSLConfig = .{
+        .secure_options = this.secure_options,
+        .request_cert = this.request_cert,
+        .reject_unauthorized = this.reject_unauthorized,
+        .client_renegotiation_limit = this.client_renegotiation_limit,
+        .client_renegotiation_window = this.client_renegotiation_window,
+        .requires_custom_request_ctx = this.requires_custom_request_ctx,
+        .is_using_default_ciphers = this.is_using_default_ciphers,
+        .low_memory_mode = this.low_memory_mode,
+        .protos_len = this.protos_len,
```

Comment:
This may be outside of the scope of this PR and I'm not asking to implement, but I was wondering why we need all of these `_len` fields? Seems like a proper type which encapsulates the intent would be better. For protocols, shouldn't the slice contain what we need, for example?